clap_derive/src/utils/doc_comments.rs

```diff
@@ -83,14 +83,64 @@ fn split_paragraphs(lines: &[&str]) -> Vec<String> {
         last_line += start + len;
 
         if len != 0 {
-            Some(merge_lines(&slice[..len]))
+            Some(process_paragraph(&slice[..len]))
         } else {
             None
         }
     })
     .collect()
 }
 
+fn process_paragraph(lines: &[&str]) -> String {
```

Comment:
If we go this route, I'd like to see test cases for this to show we are matching expected [commonmark behavior](https://spec.commonmark.org/dingus/?text=foo%5C%0Abaz%0A)